bpf/bpf_xdp.c

```diff
@@ -215,6 +220,28 @@ static __always_inline int check_v4_lb(struct __ctx_buff *ctx __maybe_unused)
 }
 #endif /* ENABLE_NODEPORT_ACCELERATION */
 
+#if defined(ENABLE_IPSEC_RPS)
+static __always_inline int ipsec_rps(const struct iphdr *ip4, const void *data_end)
+{
+	struct ip_esp_hdr *esp;
+	__u32 cpu_dest = 0;
+	__u32 hash = 0;
+
+	esp = (void *)ip4 + sizeof(*ip4);
+	if (ctx_no_room((void *)esp + sizeof(struct ip_esp_hdr), data_end))
+		return CTX_ACT_DROP;
+
+	hash = esp->spi & IPSEC_RPS_HASH_MASK;
+	esp->spi = esp->spi & IPSEC_RPS_SPI_MASK;
+	cpu_dest = hash % __NR_CPUS__;
```

Comment:
I missed that - shouldn't this be __NR_ONLINE_CPUS__?

Reply:
:facepalm: yes it definitely should, thank you for catching this.